feature-libs/checkout/occ/adapters/occ-checkout-cost-center.adapter.ts

```diff
@@ -22,27 +22,20 @@ export class OccCheckoutCostCenterAdapter implements CheckoutCostCenterAdapter {
     cartId: string,
     costCenterId: string
   ): Observable<Cart> {
-    let httpParams = new HttpParams().set('costCenterId', costCenterId);
-    /* eslint-disable max-len */
-    httpParams = httpParams.set(
-      'fields',
-      'DEFAULT,potentialProductPromotions,appliedProductPromotions,potentialOrderPromotions,appliedOrderPromotions,entries(totalPrice(formattedValue),product(images(FULL),stock(FULL)),basePrice(formattedValue,value),updateable),totalPrice(formattedValue),totalItems,totalPriceWithTax(formattedValue),totalDiscounts(value,formattedValue),subTotal(formattedValue),deliveryItemsQuantity,deliveryCost(formattedValue),totalTax(formattedValue, value),pickupItemsQuantity,net,appliedVouchers,productDiscounts(formattedValue),user'
-    );
-    // TODO(#8877): Should we improve configurable endpoints for this use case?
-
     return this.http
-      .put(
-        this.getCartEndpoint(userId) + cartId + '/costcenter',
-        {},
-        {
-          params: httpParams,
-        }
-      )
+      .put(this.getCartCostCenterEndpoint(userId, cartId, costCenterId), {})
       .pipe(this.converter.pipeable(CART_NORMALIZER));
   }
 
-  protected getCartEndpoint(userId: string): string {
-    const cartEndpoint = 'users/' + userId + '/carts/';
-    return this.occEndpoints.getEndpoint(cartEndpoint);
+  protected getCartCostCenterEndpoint(
```

Comment:
Can we also rename it to `getSetCartCostCenterEndpoint` for consistency as we changed name of the endpoint?